internal/kms/setup/setup.go

```diff
@@ -28,13 +28,29 @@ const (
 	AzureKMSURI   = "kms://azure-kms?name=%s&type=%s"
 	AzureHSMURI   = "kms://azure-hsm?name=%s"
 	GCPKMSURI     = "kms://gcp?project=%s&location=%s&keyRing=%s&protectionLvl=%s"
-	ClusterKMSURI = "kms://cluster-kms"
+	ClusterKMSURI = "kms://cluster-kms?key=%s&salt=%s"
 	AWSS3URI      = "storage://aws?bucket=%s"
 	AzureBlobURI  = "storage://azure?container=%s&connectionString=%s"
 	GCPStorageURI = "storage://gcp?projects=%s&bucket=%s"
 	NoStoreURI    = "storage://no-store"
 )
 
+// MasterSecret holds the master key and salt for deriving keys.
+type MasterSecret struct {
+	Key  []byte `json:"key"`
+	Salt []byte `json:"salt"`
+}
+
+// EncodeKey returns the url-encoded key of the MasterSecret. This encoding is necessary when sending the mastersecret as part of the KMS URI.
+func (m *MasterSecret) EncodeKey() string {
+	return base64.URLEncoding.EncodeToString(m.Key)
+}
+
+// EncodeSalt returns the url-encoded salt of the MasterSecret. This encoding is necessary when sending the mastersecret as part of the KMS URI.
+func (m *MasterSecret) EncodeSalt() string {
+	return base64.URLEncoding.EncodeToString(m.Salt)
+}
```

Comment:
```suggestion
// MasterSecret holds the master key and salt for deriving keys.
type MasterSecret struct {
	Key  []byte `json:"key"`
	Salt []byte `json:"salt"`
}

// EncodeToURI returns an URI encoding the master secret.
func (m *MasterSecret) EncodeToURI() string {
	return fmt.Sprintf(
		ClusterKMSURI,
		base64.URLEncoding.EncodeToString(m.Key),
		base64.URLEncoding.EncodeToString(m.Salt),
	}
}
```

nit: Instead of having two separate encoding methods, whose output is then manually parsed into the `ClusterKMS` format string, use one method that returns a prepared URI.

Reply:
Nice. Didn't realize that this function will only be called from code that only uses clusterKMS.